File: Medias_Moviles.m
function [mm200, mm150, mm100, mm50] = Medias_Moviles(datamm, data)

% data son los datos que queremos tratar
% datamm son los datos + 150 periodos atras para poder calcular la mm150
mm200 = NaN(size(data));
mm150 = NaN(size(data));
mm100 = NaN(size(data));
mm50 = NaN(size(data));


for i = 201:length(datamm)
    mm200(i) = mean(datamm(i-200:i));
end

for i = 151:length(datamm)
    mm150(i) = mean(datamm(i-150:i));
end

mm150 = mm150(end-(size(data,1)-1):end); %Coger los ultimos size(data,1) elementos

for i = 101:length(datamm)
    mm100(i) = mean(datamm(i-100:i));
end

mm100 = mm100(end-(size(data,1)-1):end); %Coger los ultimos size(data,1) elementos

for i = 51:length(datamm)
    mm50(i) = mean(datamm(i-50:i));
end

mm50 = mm50(end-(size(data,1)-1):end); %Coger los ultimos 251 elementos


% elimina los NaN
mm200 = mm200(~isnan(mm200));
mm150 = mm150(~isnan(mm150));
mm100 = mm100(~isnan(mm100));
mm50 = mm50(~isnan(mm50));

File: Python/Medias_Moviles.py
import numpy as np


def Medias_Moviles(datamm, data):
    # data son los datos que queremos tratar
    # datamm son los datos + 150 periodos atras para poder calcular la mm150
    datamm = np.asarray(datamm, dtype=float).ravel()
    data = np.asarray(data, dtype=float).ravel()

    N = data.size
    L = datamm.size

    # En MATLAB los arrays nacen con tamaño de 'data' y crecen hasta length(datamm)
    # dentro del bucle; aquí se reservan directamente con tamaño length(datamm).
    mm200 = np.full(L, np.nan)
    mm150 = np.full(L, np.nan)
    mm100 = np.full(L, np.nan)
    mm50 = np.full(L, np.nan)

    # OJO: replica 1:1 del original -> las ventanas son de K+1 puntos
    # (MATLAB: mean(datamm(i-200:i)) son 201 puntos, etc.)
    for i in range(201, L + 1):
        mm200[i - 1] = np.mean(datamm[i - 201:i])

    for i in range(151, L + 1):
        mm150[i - 1] = np.mean(datamm[i - 151:i])

    mm150 = mm150[L - N:]  # Coger los ultimos size(data,1) elementos

    for i in range(101, L + 1):
        mm100[i - 1] = np.mean(datamm[i - 101:i])

    mm100 = mm100[L - N:]  # Coger los ultimos size(data,1) elementos

    for i in range(51, L + 1):
        mm50[i - 1] = np.mean(datamm[i - 51:i])

    mm50 = mm50[L - N:]  # Coger los ultimos 251 elementos

    # elimina los NaN
    mm200 = mm200[~np.isnan(mm200)]
    mm150 = mm150[~np.isnan(mm150)]
    mm100 = mm100[~np.isnan(mm100)]
    mm50 = mm50[~np.isnan(mm50)]

    return mm200, mm150, mm100, mm50
